onnxruntime/core/graph/graph.cc

```diff
@@ -619,77 +618,61 @@ Graph::Graph(GraphProto* graph_proto,
   ORT_ENFORCE(graph_proto != nullptr, "graph_proto cannot be null");
   ArgNameToTypeMap name_to_type_map;
 
-  // these are all empty unless we received a graph_proto as input
-  if (graph_proto != nullptr) {
-    // Copy constant nodes _value to name_to_initial_tensor_
-    for (auto& node : graph_proto_->node()) {
-      if (node.op_type() == kConstant) {
-        const gsl::not_null<TensorProto*> tensor{graph_proto_->add_initializer()};
-        *tensor = node.attribute(0).t();
-        *(tensor->mutable_name()) = node.output(0);
-
-        // we remove the node and add it as an initializer, but still need it to appear in the
-        // graph inputs to make the ONNX checker happy. add a new input due to that.
-        auto graph_inputs = graph_proto_->mutable_input();
-
-        ValueInfoProto* value_info = graph_inputs->Add();
-        value_info->set_name(node.output(0));
-        value_info->set_doc_string("Input to represent replaced Constant node");
-
-        TypeProto t;
-        t.mutable_tensor_type()->set_elem_type(tensor->data_type());
-        auto shape = t.mutable_tensor_type()->mutable_shape();
-        for (auto dim : tensor->dims())
-          shape->add_dim()->set_dim_value(dim);
-
-        (*value_info->mutable_type()) = t;
-      }
+  for (auto& node : graph_proto_->node()) {
+    if (node.op_type() != kConstant) {
+      continue;
     }
 
```

Comment:
What's the benefit of just having the test for == kConstant without the 'continue', vs having a != test and additional 'continue' statement?